profiler/src/ProfilerEngine/Datadog.Profiler.Native/StackSnapshotResultBuffer.h

```diff
@@ -179,27 +174,30 @@ inline void StackSnapshotResultBuffer::DetermineAppDomain(ThreadID threadId, ICo
 
 inline bool StackSnapshotResultBuffer::AddFrame(std::uintptr_t ip)
 {
-    if (_currentFramesCount >= MaxSnapshotStackDepth_Limit)
-    {
-        return false;
-    }
-
-    if (_currentFramesCount == MaxSnapshotStackDepth_Limit - 1)
-    {
-        _instructionPointers[_currentFramesCount++] = 0;
-        return false;
-    }
-
-    _instructionPointers[_currentFramesCount++] = ip;
-    return true;
+    return _callstack.Add(ip);
```

Comment:
Don't we need to keep the same code that check for the max depth and add a fake frame to detect when a callstack is incomplete?

Reply:
No, fake frame or not use by the backend and discarded by when resolving frame (GetFunctionFromIP will return FALSE, so considered as native so discarded)